#4データモデリングの実施

MaekawaShun331 opened the pull request:
・予定表、候補日、出欠、コメントテーブルのマイグレーションを作成しました。
・上記テーブルとユーザテーブルのモデルファイルに対し、リレーションを定義しました。

MaekawaShun331 committed:
#4データモデリングの実施

jtakakura commented:
@MaekawaShun331 後になっての要件追加で申し訳ないのですが、 Laravel は、各テーブル必ず Auto Increment の `id` のカラムを持ち、外部との連結するカラムはそれとは別に持つというのが基本設計となっています(必須ではないのですが...)。

例えば、候補日程 (candidate)に関しては、 `candidates` テーブルの主キーは `candidateId` ではなく、 `id` としてください。

https://www.nnn.ed.nico/contents/guides/2250/content#%E5%80%99%E8%A3%9C%E6%97%A5%E7%A8%8B-candidate-%E3%81%AE%E3%83%87%E3%83%BC%E3%82%BF%E3%83%A2%E3%83%87%E3%83%AA%E3%83%B3%E3%82%B0

また、 出欠 (availabilitiy) では、 `availabilities` テーブルは、 `candidateId` と `userId` の複合主キーになっていますが、上記にならい、 `availabilities` テーブル用の `id` カラムを追加し、上記2キーは複合ユニークキーに設定してください。

https://www.nnn.ed.nico/contents/guides/2250/content#%E5%87%BA%E6%AC%A0-availabilitiy-%E3%81%AE%E3%83%87%E3%83%BC%E3%82%BF%E3%83%A2%E3%83%87%E3%83%AA%E3%83%B3%E3%82%B0

schedules テーブルの主キーは UUID 形式のままでいいのですが、カラムは上記と同じく、 `scheduleId`  ではなく、 `id` にしてください。

最後に、データベースのテーブル名、カラム名などでは一般的に大文字小文字を区別しないので、キャメルケースではなく、スネークケースを使うようにしてください。

以上、対応不明点がありましたら、質問ください。

MaekawaShun331 committed:
レビュー内容反映
・各テーブルに必ず、主キーとなるid列を持たせるよう修正（ScheduleのみUUID型のid、それ以外は自動増分id）
・主キーとなっていた外部キーをユニークキーに修正（主キーは必ずid列に修正）
・列名をキャメルケースからスネークケースに修正
・外部キーの名称を、``[参照するモデル名]_id`` となるよう修正

jtakakura commented:
@MaekawaShun331 前回指摘漏れだったのですが、テーブル一通り `timestamps` 追加をお願いします。
それ以外は大丈夫かと...。

MaekawaShun331 committed:
#4レビュー指摘内容反映
・timestampsを全てのテーブルに持たせるよう修正

jtakakura reviewed (approved):
LGTM
